### PR TITLE
Fix and simplify "rustc-watch" problem matcher

### PR DESCRIPTION
**Pull request by vemoo**
Cargo watch now always ends with "[Finished running ..." and includes some extra info. Make the pattern only match "[Finished running".

Also add "cargo watch" and "cargo watch tests" that use the "rustc-watch" problem matcher.
Change the "Build Lsp" to do "cargo check" because that's faster, and I think enought in this case, and make it use "rustc" problem matcher from the extension.

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -10,38 +10,51 @@
...
+          "relative",
+          "${workspaceRoot}/editors/code"
+        ]
       },
       "path": "editors/code/"
     },
     {
       "label": "Build Lsp",
       "type": "shell",
-      "command": "cargo build",
```
Comment (matklad) [inline .vscode/tasks.json]: `cargo build` is intentional here: it makes `F5` start an extension debug session with a freshly build analyzer

Comment (vemoo) [reply .vscode/tasks.json]: Sorry, I will undo that change

Comment (vemoo) [reply .vscode/tasks.json]: I see it now in the tasks.json. I hadn't really looked at that before, print debugging had been enough so far, but I will try it next time.

**Comment by matklad** [comment]
looks like a missing `\n` problem

bors delegate+

**Pull request merged**